### ISSUE TITLE
RuntimeError on torch.nn.parallel.DistributedDataParallel

### ISSUE THREAD

**Issue by MarsSu0618** [PyTorchLightning/pytorch-lightning#6706]
## Error Message
```
"RuntimeError: Expected to have finished reduction in the prior iteration before starting a new one. 
This error indicates that your module has parameters that were not used in producing loss. 
You can enable unused parameter detection by 
(1) passing the keyword argument `find_unused_parameters=True` to `torch.nn.parallel.DistributedDataParallel`; 
(2) making sure all `forward` function outputs participate in calculating loss. 
If you already have done the above two steps, 
then the distributed data parallel module wasn't able to locate the output tensors in the return value of your module's `forward` function. 
Please include the loss function and the structure of the return value of `forward` of your module when reporting this issue (e.g. list, dict, iterable). 
(prepare_for_backward at /opt/conda/conda-bld/pytorch_1579022060824/work/torch/csrc/distributed/c10d/reducer.cpp:514)"
```

## Environment
```
pytorch 1.4.0
pytorch-lightning 1.1.1
```
So can i upgrade version to solve it? @awaelchli 

Ref: #6517

**Comment by awaelchli** [comment]
Does this happen with one of our examples, or with your code?

Have you seen my answer here? Did you try it?
https://github.com/PyTorchLightning/pytorch-lightning/issues/6517#issuecomment-799770627

**Comment by MarsSu0618** [comment]
this happen my code on gcp ml engine.
i have tried your method that upgrade corresponding version, and still have the same error...

**Comment by edenlightning** [comment]
Can you share an example reproducible script? Or even pseudo code of what you are running?

**Comment by dalek-who** [comment]
@MarsSu0618 hey, do you solve it? I also meet this problem.
My environment:
```
pytorch-lightning          1.3.4     
torch                      1.4.0
```

 pseudo code :
```python
trainer = Trainer(
    ...
    gpus=-1,
    accelerator="ddp",
    plugins = [pl.plugins.DDPPlugin(find_unused_parameters=True)],
    ...
)
```

**Comment by dalek-who** [comment]
@SeanNaren It seems #7224 has already been merged into 1.3 stable, but I still meet this problem.
I've tried 
trainer = Trainer(
    gpus=-1,
    accelerator="ddp",
    plugins = [pl.plugins.DDPPlugin(find_unused_parameters=True)],
    ...
)
```
and
